mkosi/config.py

```diff
@@ -1381,7 +1381,7 @@ def config_parse_artifact_output_list(
     value: Optional[str], old: Optional[list[ArtifactOutput]]
 ) -> Optional[list[ArtifactOutput]]:
```

Comment:
With the below change `None` is no longer an expected output.

Reply:
The cast was technically also wrong on my end. We can still get `None` from `config_make_list_parser`